### PR TITLE
Main page

### LINKED ISSUE #32: Main Page
Create sort by component, takes array of options and emits value that is selected 

Sort by component

Create pagination component which takes initial page and emits new page after user change page

Pagination component styles

Display 6 posts on page

### PR DESCRIPTION
**Pull request by Howl404**

### CODE REVIEW

File: src/app/shared/post-cards/post-cards.service.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  private updatePaginatedData(): void {
+    if (!this.initialData) return;
+
+    const startIndex = (this.currentPage - 1) * this.itemsPerPage;
+    const endIndex = startIndex + this.itemsPerPage;
+    const paginatedData = this.fullData.slice(startIndex, endIndex);
+
+    this.dataSubject.next(paginatedData);
+  }
```
Comment (Howl404) [inline src/app/shared/post-cards/post-cards.service.ts]: Maybe will put it into PaginationService somehow

File: src/app/shared/post-cards/post-cards.service.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    this.currentSortPropertySubject.next(property);
+
+    this.fullData.sort((a, b) => {
+      if (a[property] < b[property]) return 1;
+      if (a[property] > b[property]) return -1;
+      return 0;
+    });
+
+    this.updatePaginatedData();
+  }
```
Comment (Howl404) [inline src/app/shared/post-cards/post-cards.service.ts:46]: Maybe will put it into PaginationService somehow

File: src/app/shared/table/table.component.ts [hunk 3]
```diff
@@ -8,22 +8,32 @@ import { TableService } from './table-service.service';
...
 
-  sortDirection: Observable<SortDirection> = this.tableService.sortDirection$;
+  sortDirection$: Observable<SortDirection> = this.tableService.sortDirection$;
 
-  ngOnInit(): void {
-    this.tableService.setInitialData(this.initialData);
+  lastPage = 1;
+
+  ngOnChanges(): void {
+    if (this.initialData) {
```
Comment (alesia-bulka) [inline src/app/shared/table/table.component.ts]: please use SimpleChanges to track initialData input changes

File: src/app/shared/view-switcher/view.service.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import { BehaviorSubject, Observable } from 'rxjs';
+import { ViewMode } from './view-mode.enum';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ViewService {
+  private viewModeSubject = new BehaviorSubject<ViewMode>(ViewMode.Cards);
+
+  viewMode$: Observable<number> = this.viewModeSubject.asObservable();
```
Comment (alesia-bulka) [inline src/app/shared/view-switcher/view.service.ts]:
```suggestion
  readonly viewMode$: Observable<number> = this.viewModeSubject.asObservable();
```

File: src/app/shared/table/table.component.scss [hunk 5]
```diff
@@ -48,3 +55,11 @@ tbody tr:hover {
   }
   transition: 300ms;
 }
+
+.sort-icon {
+  font-size: large;
```
Comment (alesia-bulka) [inline src/app/shared/table/table.component.scss]: use specific size values ​​to avoid different behavior in different browsers

**Review by alesia-bulka — APPROVED** [review]

**Pull request merged**